system/jlib/jbuff.cpp

```diff
@@ -64,6 +64,12 @@
 #define     CHECKREADPOS(len)  
 #endif
 
+#define CHECKBUFSIZE(len)                            \
+{                                                    \
+    size32_t newLen = curLen + (size32_t)len;        \
+    if (newLen < curLen)                             \
+        RaiseOutOfMemException(-10, newLen, curLen); \
```

Comment:
I think that error will be confusing, showing overflowed small newLen.
I would be better to explicitly show that exceeded max 32bit limit I think.